### ISSUE TITLE
MnasNet: Platform-Aware Neural Architecture Search for Mobile

### ISSUE THREAD

**Issue by sile** [sile/readings#53]
URL
-----

https://arxiv.org/abs/1807.11626

Abstract
----------

> Designing convolutional neural networks (CNN) for mobile devices is challenging because mobile models need to be small and fast, yet still accurate. Although significant efforts have been dedicated to design and improve mobile CNNs on all dimensions, it is very difficult to manually balance these trade-offs when there are so many architectural possibilities to consider. In this paper, we propose an automated mobile neural architecture search (MNAS) approach, which explicitly incorporate model latency into the main objective so that the search can identify a model that achieves a good trade-off between accuracy and latency. Unlike previous work, where latency is considered via another, often inaccurate proxy (e.g., FLOPS), our approach directly measures real-world inference latency by executing the model on mobile phones. To further strike the right balance between flexibility and search space size, we propose a novel factorized hierarchical search space that encourages layer diversity throughout the network. Experimental results show that our approach consistently outperforms state-of-the-art mobile CNN models across multiple vision tasks. On the ImageNet classification task, our MnasNet achieves 75.2% top-1 accuracy with 78ms latency on a Pixel phone, which is 1.8x faster than MobileNetV2 [29] with 0.5% higher accuracy and 2.3x faster than NASNet [36] with 1.2% higher accuracy. Our MnasNet also achieves better mAP quality than MobileNets for COCO object detection. Code is at this https URL.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) Problem Formulation
- (4) Mobile Neural Architecture Search
  - (4-1) Factorized Hierarchical Search Space
  - (4-2) Search Algorithm
- (5) Experimental Setup
- (6) Results
  - (6-1) ImageNet Classification Performance
  - (6-2) Model Scaling Performance
  - (6-3) COCO Object Detection Performance
- (7) Ablation Study and Discussion
  - (7-1) Soft vs. Hard Latency Constraint
  - (7-2) Disentangling Search Space and Reward
  - (7-3) MnasNet Architecture and Layer Diversity
- (8) Conclusion

BibTeX
------

```
@inproceedings{tan2019mnasnet,
  title={Mnasnet: Platform-aware neural architecture search for mobile},
  author={Tan, Mingxing and Chen, Bo and Pang, Ruoming and Vasudevan, Vijay and Sandler, Mark and Howard, Andrew and Le, Quoc V},
  booktitle={Proceedings of the IEEE Conference on Computer Vision and Pattern Recognition},
  pages={2820--2828},
  year={2019}
}
```

See Also
----------

- GitHub: https://github.com/tensorflow/tpu/tree/master/models/official/mnasnet

**Comment by sile** [comment]
> To summarize, our main contributions are as follows:
> 1. We introduce a multi-objective neural architecture search approach that optimizes both accuracy and realworld latency on mobile devices.
> 2. We propose a novel factorized hierarchical search space to enable layer diversity yet still strike the right balance between flexibility and search space size.
> 3. We demonstrate new state-of-the-art accuracy on both ImageNet classification and COCO object detection under typical mobile latency constraints.
